app/models/service_provider_identity.rb

```diff
@@ -57,6 +57,11 @@ def friendly_name
     sp_metadata[:friendly_name]
   end
 
+  def all_email_and_single_email_requested?
```

Comment:
I don't think this is the behavior we're expecting. A partner would only request either `email` or `all_emails` in most cases. We only want to link the email if they requested a singular `email`, since this is when we show the option for the user to choose which email should be shared.

Reply:
Oh the AC in the ticket said we were accounting for all emails and email requested. ```Email shared with partner defaults to last-used email if user never sees email change option (e.g. both email and all_emails requested)```

Reply:
https://cm-jira.usa.gov/browse/LG-15251

Reply:
But that may be simpler where it covers both cases? just was under the impression that it was for both not just all emails.

Reply:
"Both" there is referring to two separate scenarios (either the partner requests `email` or the partner requests `all_emails`), not a scenario where both are requested at the same time. But I agree that wording is confusing.

Reply:
Sorry, there was an error in my last comment. The two scenarios are:

- The partner doesn't request any email (there is neither `email` nor `all_emails` in the requested attributes)
- The partner requests all emails (`all_emails` in requested attributes)

The goal is to make sure we only link / return a selected email if the user had the option to choose a selected email. The "Change" link only shows up if the requested attributes includes `email` and does not include `all_emails`.